website/docs/cdktf/concepts/terraform-cloud.mdx

```diff
@@ -0,0 +1,112 @@
+---
+page_title: Terraform Cloud
+description: >-
+  Use Terraform Cloud to keep your infrastructure state, apply policies, and execute infrastructure changes.
+---
+
+# Terraform Cloud
+
+Terraform Cloud is HashiCorp’s managed service offering. It eliminates the need for unnecessary tooling and documentation for practitioners, teams, and organizations to use Terraform in production.
+
+Provision infrastructure in a remote environment that is optimized for the Terraform workflow.
+
+You can use [these learn guides to learn more about Terraform Cloud](https://learn.hashicorp.com/collections/terraform/cloud-get-started).
+
+## Why Terraform Cloud & Terraform Enterprise?
+
+CDK for Terraform supports [Terraform Cloud](https://cloud.hashicorp.com/products/terraform), the managed service offering, and [Terraform Enterprise](/enterprise), the self-hosted distribution of Terraform Cloud.
+
+Both Terraform Cloud and Terraform Enterprise can be used with CDKTF to [plan and execute infrastructure builds](/cloud-docs/run/remote-operations), [enforce organization-wide policies](/cloud-docs/sentinel), [estimate costs](/cloud-docs/cost-estimation), and [manage variables](/cloud-docs/workspaces/variables).
+
+You may want to use Terraform Cloud when you need either of those functionalities or if you want a simple and reliable way to store the Terraform state. The default of writing the state into a local file means the risk of loosing your state by accident (the state contains sensitive information and should not be kept in version control), so we recommend Terraform Cloud as a way to persist the data.
+
+## Set up CDKTF with Terraform Cloud
```

Comment:
Okay this is a good start! I'm thinking about how to connect the dots for users who are new:
- Don't you have to run cdktf init (or something to log in to TFC from your terminal?) to connect to TFC? 
- Maybe we should explain briefly there are three types of workflows (CLI-driven, VCS-driven, and API-driven) that you can use when working with TFC workspaces. And then say that these docs are going to explain how to set up with VCS and CLI. 
- It feels like there are kind of two use cases for CLI that we're trying to cover here. For CLI you can either connect an existing project --> that's when you need to manually like create the workspace and add the remote backend to your config. OR you can connect a new project by running 'cdktf init' and it does it for you automatically.  Yes?

So maybe we revise this to roughly something like:

## Set up CDKTF with Terraform Cloud

After you sign up for a Terraform Cloud account (link), you will connect your CDKTF project to one or more Terraform Cloud workspaces. 

Workspaces function like working directories for distinct Terraform configurations (link to workspaces docs). The workspace is associated with a Terraform configuration and its associated state file. It can also contain variables that you can use for credential management. etc. etc.By default, workspaces are created in Remote Execution Mode (link), which enables Terraform Cloud to store Terraform state and run Terraform operations remotely.

Explain about the different workflows you can use (link to TFC pages on these). 

### CLI-driven Workflow

You can configure the Terraform Cloud CLI-driven workflow for both new and existing CDKTF projects. 

#### Connect New CDKTF Projects

For new projects, run `cdktf init` without passing the `--local` flag. CDKTF will use your stored [Terraform Cloud](/cloud) credentials to create a new workspace. If you have no stored Terraform Cloud credentials, CDKTF will ask you to login.

After you login, CDKTF creates a new scaffolded project that is configured to use Terraform state with a [remote backend](/language/settings/backends/remote). **Will TFC automatically st the workspace to remote execution??**

#### Connect Existing CDKTF Projects
For existing projects, do the following:
1. Create a Terraform Cloud workspace with the CLI-driven workflow.
2. Add the `RemoteBackend` to your CDKTF program. (Example)
3. Run `cdktf init` so that you can actually log into TFC and connect with your credentials?

### VCS-driven Workflow

Something about how you should synthesize your configuration and set up your TFC workspace to point to that directory :) Useful for CI/CD pipelines (link to that Deployment patterns page).

Reply:
I'm not sure if this makes it better, variables etc are not covered / don't fit into this structure. 

> Don't you have to run cdktf init (or something to log in to TFC from your terminal?) to connect to TFC?

Yeah, we did not cover `cdktf login`, will add that one

> Maybe we should explain briefly there are three types of workflows (CLI-driven, VCS-driven, and API-driven) that you can use when working with TFC workspaces. And then say that these docs are going to explain how to set up with VCS and CLI.

I'm not sure if this will not convolute the page, I'd like to focus on what's necessary to understand to connect to TFC

> It feels like there are kind of two use cases for CLI that we're trying to cover here. For CLI you can either connect an existing project --> that's when you need to manually like create the workspace and add the remote backend to your config. OR you can connect a new project by running 'cdktf init' and it does it for you automatically. Yes?

Yes, having these two after another was kind of intentional, so that it surfaces what happens on cdktf init so folks are not surprised

Reply:
Now I think I understand what you meant, let me try it

Reply:
Redid it 👍